### PR TITLE
Fix miscellaneous grammar issues and perform light editing.

### PR DESCRIPTION
**Pull request by david-mitchell**

### CODE REVIEW

**Comment by jkotas** [comment]
@richlander Could you please take a look?

**Comment by mmitche** [comment]
PR test is good (issue fixed in recent commit)

**Comment by richlander** [comment]
@david-mitchell Thanks for your updates to the doc.

Two things:
- Can you squash all these commits into one?
- I'm less keen on the "runtime" -> "run time" change. That may be more correct, but we never write it that way. You can say that this is an example of a "regional dialect".

**Comment by david-mitchell** [comment]
@richlander I'd be happy to squash the commits.

As to the "runtime" -> "run time" changes, I did this because that already appeared to be the convention within the document. Should *all* occurrences of "run time" be changed to "runtime"? (including the ones I did not originally write)

**Comment by richlander** [comment]
@david-mitchell You are right. Thanks for pointing that out.

Then, just one request, which is to squash, and then I'll merge.

BTW: We're always happy to receive PRs as a large set of commits. No issue there. Just squashing for the official history.

File: Documentation/intro-to-clr.md [hunk 1]
```diff
@@ -35,55 +35,55 @@ Now that we have basic idea what the CLR is, it is useful to back up just a bit
...
+At any time some other thread of execution might perform an allocation requires a garbage collection.  The exact sequence of operations across concurrently executing threads is non-deterministic.  We can't tell exactly what one thread will be doing when another thread requests an allocation that will trigger a GC.  Thus, GCs can really happen any time.  Now the CLR does NOT need to respond _immediately_ to another thread's desire to do a GC, so the CLR has a little "wiggle room" and doesn't need to track GC references at _all_ points of execution, but it _does_ need to do so at enough places that it can guarantee "timely" response to the need to do a GC caused by an allocation on another thread.
```
Comment (akoeplinger) [inline Documentation/intro-to-clr.md]: `might perform an allocation requires a garbage collection` <-- missing a "that" in there. LGTM otherwise.

**Comment by david-mitchell** [comment]
Done.

**Comment by richlander** [comment]
Thanks again @david-mitchell .

**Pull request merged**